File: apzkr-pzpi-21-4-bondarenko-taras/Task4-MobileClient/CargoTruckMobileClient/app/src/main/java/com/example/cargotruckmobile/Network/CargoService.kt
package com.example.cargotruckmobile.Network

import com.example.cargotruckmobile.Models.Cargo
import com.example.cargotruckmobile.Models.CargoCreateDto
import com.example.cargotruckmobile.Models.CargoListResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface CargoService {

    @GET("cargo/get/sender{id}")
    suspend fun getCargoBySender(@Path("id") id: String): List<Cargo>

    @GET("cargo/get/receiver{id}")
    suspend fun getCargoByReceiver(@Path("id") id: String): List<Cargo>

    @GET("cargo?pageNumber=1&pageSize=100")
    suspend fun getCargoLogist(): CargoListResponse

    @GET("cargo/get/{id}")
    suspend fun getCargo(@Path("id") id: String): Cargo

    @POST("cargo/create")
    suspend fun createCargo(@Body cargo: CargoCreateDto): Response<Void>

    @PUT("cargo/update/notice")
    suspend fun updateNotice(
        @Query("cargoId") cargoId: String,
        @Query("newNoticeId") newNoticeId: String
    ): Response<Void>

    @GET("cargo/get?pageNumber=1&pageSize=100")
    suspend fun getCargoPage(): CargoListResponse


}

File: apzkr-pzpi-21-4-bondarenko-taras/Task3-WebClient/CargoTruckWebClient/src/Api/CargoService.js
import { ApiClient } from './ApiClient';

export const CargoService = {
  getCargoBySender: async (id) => {
    const response = await ApiClient.get(`cargo/get/sender${id}`);
    return response.data;
  },
  getCargoByReceiver: async (id) => {
    const response = await ApiClient.get(`cargo/get/receiver${id}`);
    return response.data;
  },
  getCargo: async (id) => {
    const response = await ApiClient.get(`cargo/get/${id}`);
    return response.data;
  },
  updateCargo: async (cargo) => {
    const response = await ApiClient.put('cargo/update', cargo);
    return response.data;
},
  getCargoPage: async () => {
    const response = await ApiClient.get(`cargo/get?pageNumber=1&pageSize=100`);
    return response.data.items;
  },
  createCargo: async (cargo) => {
    const response = await ApiClient.post('cargo/create', cargo);
    return response;
  },
  updateNotice: async (cargoId,newNoticeId) => {
    const response = await ApiClient.put(`cargo/update/notice?cargoId=${cargoId}&newNoticeId=${newNoticeId}`);
    return response;
  },
  getCargoStatus: async () => {
    const response = await ApiClient.get('cargo?pageNumber=1&pageSize=100');
    return response.data.items;
  },
  findContainer: async (lat,lng) => {
    const response = await ApiClient.get(`container/find/${lat}/${lng}`);
    return response.data;
  },
  deleteCargo: async (id) => {
    const response = await ApiClient.delete(`cargo/delete/${id}`);
    return response;
  }

};
